src/qdarts/tunneling_simulator.py

```diff
@@ -3,260 +3,288 @@
-            #add noise
-            eps = eps + slow_noise[i:i+1]
+            terms, neighbour_prev, neighbour_next, _ = sensor_state[sensor_id]
+
+            # compute sensor detuning between every state and their neighbour
+            eps_prev = np.abs(np.diag(H)[terms] - np.diag(H)[neighbour_prev])
+            eps_next = np.abs(np.diag(H)[terms] - np.diag(H)[neighbour_next])
+            eps = np.minimum(eps_prev, eps_next)
+            # add noise
+            eps = eps + slow_noise[i : i + 1]
             if self.fast_noise_var > 0:
-                fast_noise = np.random.randn(*eps.shape)*np.sqrt(self.fast_noise_var)
+                fast_noise = np.random.randn(*eps.shape) * np.sqrt(self.fast_noise_var)
                 eps += fast_noise
-            eps *=beta 
-            
-            #we approximate the logistic peak of g with the peak of a normal distribution of same width
-            #todo: we can fully go back to the logistic peak
-            var_logistic = (1/0.631*self.peak_width_multiplier)**2
-            norm_pdf = lambda x, mu,var: 1/np.sqrt(2*np.pi*var)*np.exp(-(x-mu)**2/(2*var))
-            gs[sensor_id] = self.g_max*4*norm_pdf(0,eps, var_logistic)
+            eps *= beta
+
+            # we approximate the logistic peak of g with the peak of a normal distribution of same width
+            # todo: we can fully go back to the logistic peak
+            var_logistic = (1 / 0.631 * self.peak_width_multiplier) ** 2
+
+            def norm_pdf(x, mu, var):
+                return (
+                    1 / np.sqrt(2 * np.pi * var) * np.exp(-((x - mu) ** 2) / (2 * var))
+                )
+
+            gs[sensor_id] = self.g_max * 4 * norm_pdf(0, eps, var_logistic)
         return gs
+
     def sample_sensor_equilibrium(self, v, H, mixed_state, sensor_state, beta):
         results = np.zeros(len(self.sensor_dot_ids))
         gs = self._precompute_g(v, H, sensor_state, beta)
-        
+
         for i, sensor_id in enumerate(self.sensor_dot_ids):
-            terms,neighbour_prev, neighbour_next, _ = sensor_state[sensor_id]
+            terms, neighbour_prev, neighbour_next, _ = sensor_state[sensor_id]
             g = gs[sensor_id]
             p = np.diag(mixed_state)[terms]
-            results[i] = np.sum(p*g)/np.sum(p)
-        var_logistic = (1/0.631*self.peak_width_multiplier)**2
-        scale = self.g_max*self.signal_noise_scale*4/np.sqrt(2*np.pi*var_logistic)
-        results += scale*np.random.randn(len(results))
+            results[i] = np.sum(p * g) / np.sum(p)
+        var_logistic = (1 / 0.631 * self.peak_width_multiplier) ** 2
+        scale = (
+            self.g_max * self.signal_noise_scale * 4 / np.sqrt(2 * np.pi * var_logistic)
+        )
+        results += scale * np.random.randn(len(results))
         return results
-        
-    def sample_sensor_configuration(self, sampled_configuration, v, H, mixed_state, sensor_state, beta):
+
+    # TODO: This function is identical to the previous one ?
```

Comment:
it is not. this computes the sensor response to a specific charge configuration, while the other function computes the thermalised average over all configurations

Reply:
I get it now, I did have to add the `var_logistic` from `sample_sensor_configuration` since it was missing here. In any case there is so much similarity between the functions, that it's worth thinking about some refactoring in order to minimize code duplication.